### PR TITLE
Added possibility to share multiple calendars

### PR DESCRIPTION
**Pull request by Ich5003**
This PR implements a function requested in #708 
You can pass multiple share tokens split by a . to embed multiple calendars.
All calendars are shown in the default color at the moment. This may be impoved.

### CODE REVIEW

File: js/app/controllers/calcontroller.js [hunk 1]
```diff
@@ -140,17 +140,28 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
...
+                        $scope.$apply();
+                    }
+                }).catch((reason) => {
+                    angular.element('#header-right').css('display', 'none');
+                    angular.element('#emptycontent-container').css('display', 'block');
+                });
+            });
 		}
 
+        $scope.integration = function () {
```
Comment (georgehrke) [inline js/app/controllers/calcontroller.js]: Why did you move this here?

File: js/app/controllers/calcontroller.js [hunk 2]
```diff
@@ -140,17 +140,28 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
...
-				$scope.$apply();
-			}).catch((reason) => {
-				angular.element('#header-right').css('display', 'none');
-				angular.element('#emptycontent-container').css('display', 'block');
-			});
+		    $scope.calendars = [];
+		    var tokens = constants.publicSharingToken.split(".");
+		    var finished = 0;
+		    tokens.forEach(function(token){
+                $scope.calendarsPromise = CalendarService.getPublicCalendar(token).then(function(calendar) {
```
Comment (georgehrke) [inline js/app/controllers/calcontroller.js]: Please ident with tabs instead of spaces

File: js/app/controllers/calcontroller.js [hunk 3]
```diff
@@ -140,17 +140,28 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
...
+                    finished++;
+                    $scope.calendars.push(calendar);
+                    if(finished >= tokens.length){
+                        is.loading = false;
+                        // TODO - scope.apply should not be necessary here
+                        $scope.$apply();
+                    }
+                }).catch((reason) => {
+                    angular.element('#header-right').css('display', 'none');
```
Comment (georgehrke) [inline js/app/controllers/calcontroller.js]: This error handling needs to be adapted.
You don't want to show the empty-container error page if one out of ten calendars fail. 
You would want some kind of Notification though. You can use `OC.Notification.show()` for that.

File: js/app/controllers/calcontroller.js [hunk 4]
```diff
@@ -140,17 +140,28 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
...
+                    angular.element('#emptycontent-container').css('display', 'block');
+                });
+            });
 		}
 
+        $scope.integration = function () {
+            return '<iframe width="400" height="215" src="' + $scope.$parent.root + 'embed/' + constants.publicSharingToken + '"></iframe>';
```
Comment (georgehrke) [inline js/app/controllers/calcontroller.js]: Not sure if angular handles escaping here.
What if you pass html as a parameter for `constants.publicSharingToken`?

This might result in an html injection.

Can you please check if its properly escaped?

File: js/app/controllers/calcontroller.js (hunk 1 above)
Comment (Ich5003) [reply js/app/controllers/calcontroller.js]: Moved here, because otherwise the embed code will only contain one of the public sharing tokens.

File: js/app/controllers/calcontroller.js (hunk 3 above)
Comment (Ich5003) [reply js/app/controllers/calcontroller.js]: Will check.

File: js/app/controllers/calcontroller.js (hunk 4 above)
Comment (Ich5003) [reply js/app/controllers/calcontroller.js]: Will check

**Comment by Ich5003** [comment]
I messed up commits, which automatically closed this PR.
Had to create a new one, see #870
